jerry-core/api/jerry.c

```diff
@@ -254,6 +255,44 @@ jerry_gc (void)
   ecma_gc_run (JMEM_FREE_UNUSED_MEMORY_SEVERITY_LOW);
 } /* jerry_gc */
 
+#ifdef JMEM_STATS
+void jmem_heap_get_stats (jmem_heap_stats_t *);
+#endif
+/**
+ * Get heap memory stats.
+ *
+ * @return true - get the heap stats successful
+ *         false - otherwise. Usually it is because the MEM_STATS feature is not enabled.
+ */
+bool
+jerry_get_memory_stats (jerry_heap_stats_t *out_stats_p) /**< [out] heap memory stats */
+{
+#ifdef JMEM_STATS
+  if (out_stats_p == NULL)
+  {
+    return false;
+  }
+
+  jmem_heap_stats_t jmem_heap_stats = {0};
+  jmem_heap_get_stats (&jmem_heap_stats);
+
```

Comment:
@zherczeg Personally, I was also surprised about your concerns w.r.t. version mismatches when dynamic linking, because – as @jiangzidong also pointed out – the rest of the API is not really taking that concern into account at all. Breaking ABI changes have been made in the past without anyone complaining that their dynamically linked libraries became incompatible.

I also still feel it makes the API pretty ugly and heavy if you have to pass in versions all over the place. I'd rather have a over-arching "JerryScript version" that the caller can use to check for compatibility.